R/zchunk_LA101.ag_FAO_R_C_Y.R

```diff
@@ -40,79 +44,141 @@ module_aglu_LA101.ag_FAO_R_C_Y_DISABLED <- function(command, ...) {
+      mutate(value = value * CONV_TON_MEGATON)  %>%                                                                # Convert from tons to Mt
+      group_by(GCAM_region_ID, GCAM_commodity, year) %>%                                                           # Group by region, commodity, year
+      summarize(value=sum(value)) %>%                                                                              # Aggregate then map to appropriate data frame
+      ungroup() %>%                                                                                                # Ungroup before complete
+      complete(GCAM_region_ID = unique(iso_GCAM_regID$GCAM_region_ID),
+                GCAM_commodity, year, fill = list(value = 0)) ->                                                   # Fill in missing region/commodity combinations with 0
+      L101.ag_Food_Mt_R_C_Y
+    
+    # Process FAO food consumption data (Pcal): remove unnecessary columns, convert units, aggregate to region and commodity
+    L100.FAO_ag_Food_t %>% 
+      select(-countries, -country.codes, -item.codes, -element, -element.codes) %>%                                # Remove unnecessary columns
+      left_join_error_no_match(iso_GCAM_regID, by=c("iso"))  %>%                                                   # Map in ISO codes
+      left_join(FAO_ag_items_cal_SUA, by=c("item"))  %>%                                                           # Map in GCAM commodities
+      filter(GCAM_commodity != "NA") %>%                                                                           # Remove commodities not included in GCAM
+      mutate(value = value * Mcal_t * CONV_MCAL_PCAL)  %>%                                                         # Convert from tons to Pcal
+      group_by(GCAM_region_ID, GCAM_commodity, year) %>%                                                           # Group by region, commodity, year
+      summarize(value=sum(value)) %>%                                                                              # Aggregate then map to appropriate data frame
+      ungroup() %>%                                                                                                # Ungroup before complete
+      complete(GCAM_region_ID = unique(iso_GCAM_regID$GCAM_region_ID),
+                GCAM_commodity, year, fill = list(value = 0)) ->                                                   # Fill in missing region/commodity combinations with 0
+      L101.ag_Food_Pcal_R_C_Y
+    
+    # Calculate average caloric content of consumed commodities (kcal/g)
+    L101.ag_Food_Pcal_R_C_Y %>% 
+      left_join(L101.ag_Food_Mt_R_C_Y, by = c("GCAM_region_ID", "GCAM_commodity", "year")) %>%                  # Join food in Mt to food in Pcal
+      mutate(value = ifelse(value.y == 0, 1, value.x / value.y))  %>%                                           # Calculate average caloric content, set NA values to 1
```

Comment:
@kvcalvin I'll make this change but FYI - you *never* want to use `ifelse` in a pipeline like this; use `if_else` (the dplyr version) instead.